### PR TITLE
Adding unit_testing examples from presentation

### PR DESCRIPTION
**Pull request by Arun-bioinformatics**
- Updated the main README.md to specify the inclusion of the new content
- Added a second README.md inside the Unit_testing_python_2026 with some explanation on how to run pytest.

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/code_school/19)

## Summary by CodeRabbit

* **Documentation**
  * Added a "2026 additions" section and expanded learning-session README with a unit testing presentation, usage instructions, prerequisites and test commands.

* **New Features**
  * Added utilities for ClinVar significance lookup, DNA GC-content calculation and BED file length computation.

* **Tests**
  * Added comprehensive unit tests covering normal cases, edge cases and error handling for all new utilities.

### CODE REVIEW

File: learning-sessions/Unit_testing_python_2026/example_code/bin/clinvar_utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    Raises
+    ------
+    requests.exceptions.RequestException
+        If the API call fails.
+    ValueError
+        If the Variation ID is not found or data is malformed.
+    """
+    # ClinVar API endpoint for variant data
+    url = ("https://api.ncbi.nlm.nih.gov/variation/v0/beta/"
+           f"clinical-significance/variation/{variation_id}")
```
[the remark this answers is not in the document]
Comment (Arun-bioinformatics) [reply learning-sessions/Unit_testing_python_2026/example_code/bin/clinvar_utils.py:28]: It doesn't really matter whether the function works or not. This is just supposed to be an example for unit testing code.

File: learning-sessions/Unit_testing_python_2026/example_code/tests/test_script.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    def test_rounding(self):
+        """Test that the result is rounded correctly."""
+        # GC content of "GAT" is 1/3 = 33.3333...%
+        assert calculate_gc_content("GAT") == 33.33
+
+    def test_invalid_characters(self):
+        """Test that a ValueError is raised for sequences with invalid characters."""
+        with pytest.raises(ValueError, match="Sequence contains invalid characters. "
+                           "Only A, T, G, C are allowed."):
+            calculate_gc_content("ATGCX")
```
[the remark this answers is not in the document]
Comment (Arun-bioinformatics) [reply learning-sessions/Unit_testing_python_2026/example_code/tests/test_script.py]: If I apply the proposed fix, I get a SyntaxWarning: invalid escape sequence '\.'. I really don't want that

**Review by Arun-bioinformatics — COMMENTED** [review]
> @Arun-bioinformatics made 2 comments.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/code_school/19)* status: 0 of 10 files reviewed, 2 unresolved discussions (waiting on RSWilson1).

**Review by Arun-bioinformatics — COMMENTED** [review]
Nitpick comment addressed

> @Arun-bioinformatics made 1 comment.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/code_school/19#-:-OmOOiNE0hOaXowr0tJI:bksfgdj)* status: 0 of 10 files reviewed, 1 unresolved discussion (waiting on RSWilson1).

File: learning-sessions/Unit_testing_python_2026/example_code/tests/test_script.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Arun-bioinformatics) [reply learning-sessions/Unit_testing_python_2026/example_code/tests/test_script.py]: Solution applied

**Review by Arun-bioinformatics — COMMENTED** [review]
> @Arun-bioinformatics made 1 comment.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/code_school/19)* status: 0 of 10 files reviewed, 1 unresolved discussion (waiting on RSWilson1).

**Review by Arun-bioinformatics — COMMENTED** [review]
> @Arun-bioinformatics resolved 1 discussion.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/code_school/19)* status: 0 of 10 files reviewed, all discussions resolved (waiting on RSWilson1).

**Review by RSWilson1 — APPROVED** [review]
> @RSWilson1 reviewed 10 files and all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/code_school/19)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on Arun-bioinformatics).
